mitodl/mit-learn
# Refactor user lists and learning paths hooks out of learning resources

### What are the relevant tickets?





N/A

### Description (What does it do?)


Refactors the learning resource API hooks to pull out Learning Paths and User lists. This align the cache key structure with the API path structure / data models, keying `["learningPaths", ...]` and `["userLists", ...]` at the top level. Much of the data on the site relates to learning resources, so the learning resources API handling was getting large with relationships as children of learning resources.

This identified a lot of the user list code as unused. It is commented out in this PR so we can first determine whether it is worth keeping for any upcoming usage.

### How can this be tested?


Non-functional change. Test should pass and everything should work as previously, with particular attention to user list and learning path memberships and their action buttons on resource cards and list items; ensuring that these behave as expected as we add and remove from lists and navigate between screens (homepage carousels, search results, dashboard user list details and admin learning paths).

### Additional Context



Refactored to lay things out more clearly for https://github.com/mitodl/hq/issues/6032, which introduces memberships that specify the relationships between learning resources and lists (the membership lists are on the learning path and user lists APIs).
